src/Explorer/Controls/SmartUi/SmartUiComponent.tsx

```diff
@@ -86,24 +101,41 @@ export interface SmartUiDescriptor {
 }
 
 /************************** Component implementation starts here ************************************* */
+export interface SmartUiInput {
+  value: InputType;
+  hidden?: boolean;
+  disabled?: boolean;
+}
 
 export interface SmartUiComponentProps {
   descriptor: SmartUiDescriptor;
-  currentValues: Map<string, InputType>;
+  currentValues: Map<string, SmartUiInput>;
   onInputChange: (input: AnyInput, newValue: InputType) => void;
+  onError: (hasError: boolean) => void;
+  disabled: boolean;
 }
 
 interface SmartUiComponentState {
   errors: Map<string, string>;
 }
 
 export class SmartUiComponent extends React.Component<SmartUiComponentProps, SmartUiComponentState> {
+  private shouldCheckErrors = true;
   private static readonly labelStyle = {
     color: "#393939",
     fontFamily: "wf_segoe-ui_normal, 'Segoe UI', 'Segoe WP', Tahoma, Arial, sans-serif",
     fontSize: 12,
   };
 
+  componentDidUpdate(): void {
+    if (!this.shouldCheckErrors) {
```

Comment:
This looks strange. What is the purpose of this logic?
Should `shouldCheckErrors` be moved into `SmartUiComponentState`?

Reply:
We have some errors defined in smartUiComponent. Right now, only for the spinner number input, where it checks if the input is a number and is within range, else shows a message bar with the error next to the input. More can be added in the future
The purpose of props.onError is to set a state in the parent component so that the save button is disabled when errors are present in the smartUiComponent.
since the parent state is modified in the onComponentChange of the child, this will enter an infinite loop if not for a control variable which decides when the props.onError() needs to be called. that is the role of "shouldCheckErrros"